File: sol/solution/0001-0100/0026/Solution.rb
# @param {Integer[]} nums
# @return {Integer}
def remove_duplicates(nums)
    return 0 if nums.empty?
    write = 1
    (1...nums.length).each do |i|
        if nums[i] != nums[write - 1]
            nums[write] = nums[i]
            write += 1
        end
    end
    write
end

File: sol/solution/0001-0100/0026/Solution.php
class Solution {
    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function removeDuplicates(&$nums) {
        if (count($nums) === 0) return 0;
        $k = 0;
        for ($i = 1; $i < count($nums); $i++) {
            if ($nums[$i] !== $nums[$k]) {
                $k++;
                $nums[$k] = $nums[$i];
            }
        }
        return $k + 1;
    }
}